test/functional/options/defaults_spec.lua

```diff
@@ -328,3 +329,64 @@ describe('XDG-based defaults', function()
     end)
   end)
 end)
+
+
+describe('stdpath()', function()
+  before_each(function()
+    clear({env={
+      XDG_CONFIG_HOME='cat',
```

Comment:
The code is basically string manipulation; it doesn't know anything about separators.  And I don't want to write tests that test `get_xdg_home()` and friends; those tests belong someplace else.